vizro-core/tests/unit/vizro/models/test_types.py

```diff
@@ -148,12 +163,28 @@ def test_decorated_graph_function(self):
         model = Model(function=decorated_graph_function(data_frame=None))
         assert model.function() == go.Figure()
 
+    def test_decorated_df_standard_case(self, gapminder):
+        fig = decorated_graph_function_px(gapminder)
+        assert len(fig.data) > 0
+        assert fig.__class__ == _DashboardReadyFigure
+
+    def test_decorated_df_str(self):
+        fig = decorated_graph_function_px("gapminder")
+        assert fig == _DashboardReadyFigure()
+        assert len(fig.data) == 0
+
```

Comment:
Are these tests at the correct place / can you remind me again why we devided to test these?

Reply:
to cover this block of code:
```
                if isinstance(captured_callable["data_frame"], str):
                    # Enable running e.g. px.scatter("iris") from the Python API. Don't actually run the function
                    # because it won't get work as there's no data. It's vital we don't fetch data from the data manager
                    # yet either, because otherwise all lazy data will be loaded before the dashboard is started.
                    # This case is not relevant for the JSON/YAML API, which is handled separately through validation of
                    # CapturedCallable.
                    fig = _DashboardReadyFigure()
                else:
                    # Standard case for px.scatter(df: pd.DataFrame).
                    fig = func(*args, **kwargs)
                    fig.__class__ = _DashboardReadyFigure

Reply:
Hm I see, good point. Is there a reason why we do not use the `Model`. I was a bit confused, because these tests are grouped under `TestModelFieldPython` and there was no `Model` in your tests.

Aside, what is `assert len(fig.data)` testing?

Reply:
assert len(fig.data)  is testing whether there are data attached